helix-term/src/commands.rs

```diff
@@ -2431,6 +2434,48 @@ fn file_picker_in_current_directory(cx: &mut Context) {
     cx.push_layer(Box::new(overlayed(picker)));
 }
 
+fn open_or_focus_explorer(cx: &mut Context) {
+    cx.callback = Some(Box::new(
+        |compositor: &mut Compositor, cx: &mut compositor::Context| {
+            if let Some(editor) = compositor.find::<ui::EditorView>() {
+                match editor.explorer.as_mut() {
+                    Some(explore) => explore.content.focus(),
+                    None => match ui::Explorer::new(cx) {
+                        Ok(explore) => editor.explorer = Some(overlayed(explore)),
+                        Err(err) => cx.editor.set_error(format!("{}", err)),
+                    },
+                }
+            }
+        },
+    ));
+}
+
+fn reveal_current_file(cx: &mut Context) {
+    cx.callback = Some(Box::new(
+        |compositor: &mut Compositor, cx: &mut compositor::Context| {
+            if let Some(editor) = compositor.find::<ui::EditorView>() {
+                (|| match editor.explorer.as_mut() {
```

Comment:
why closure & iife?

In L#2441 direct match ```match editor.explorer.as_mut() {``` used.

Reply:
The IIFE is so that I can use the `?` at L#2460, since there's no equivalent of Haskell's do-notation in Rust yet.

If I unwrap the IIFE, it becomes quite nested, and it will get worse as more Result-returning-function calls are added to the block.

The unwrap version for reference:
```rs
let result = match editor.explorer.as_mut() {
    Some(explore) => explore.content.reveal_current_file(cx),
    None => match ui::Explorer::new(cx) {
        Err(err) => Err(err),
        Ok(explorer) => {
            editor.explorer = Some(overlayed(explorer));
            let explorer = editor.explorer.as_mut().unwrap();
            explorer.content.reveal_current_file(cx)
        }
    },
};
if let Err(err) = result {
    cx.editor.set_error(err.to_string())
}
```
For now, I don't mind unwrapping it since there's not too much nesting yet, but it will difficultize code addition in the future.